### PR TITLE
Test all beats can setup their template

### PR DESCRIPTION
**Pull request by exekias**
This will ensure we don't introduce errors in our fields definitions

### CODE REVIEW

File: auditbeat/tests/system/test_base.py [hunk 1]
```diff
@@ -23,3 +25,24 @@ def test_start_stop(self):
         assert self.log_contains("Setup Beat: auditbeat")
         assert self.log_contains("auditbeat start running")
         assert self.log_contains("auditbeat stopped")
+
+    def test_template(self):
```
Comment (andrewkroh) [inline auditbeat/tests/system/test_base.py]: I think this should have a `@unittest.skipUnless(INTEGRATION_TESTS, "integration test")`?

**Comment by ruflin** [comment]
Seems this failed for metricbeat on travis:

```
======================================================================
FAIL: Test that the template can be loaded with `setup --template`
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/metricbeat/tests/system/test_base.py", line 48, in test_template
    assert exit_code == 0
AssertionError
```

@exekias Thanks for taking this one over.

**Comment by exekias** [comment]
it seems `TEST_ENVIRONMENT` is always `false` for packetbeat, will need to add a `docker-compose.yml` for this

**Comment by ruflin** [comment]
@exekias Should we skip packetbeat for now in this PR and add it in a follow up PR?

**Comment by exekias** [comment]
Looks good to me, I'm afraid I have little bandwidth at the moment, could get something by the end of the week, if everything goes well, but merging now sounds safer

**Comment by ruflin** [comment]
@exekias To get this in now (which I think we should do) you need to remove the packetbeat commit again to get it to green.

**Comment by exekias** [comment]
Applied your comment and rebased with master to get latest tests fixes

File: metricbeat/module/elasticsearch/_meta/env [hunk 2]
```diff
@@ -1,2 +1,2 @@
-ELASTICSEARCH_HOST=elasticsearch
-ELASTICSEARCH_PORT=9200
+ES_HOST=elasticsearch
```
Comment (ruflin) [inline metricbeat/module/elasticsearch/_meta/env]: Is there a specific reason you changed this? I kept it identical to the module name as I initially intended to do some auto generation here. Will merge it as is but curious if there were some issues with the longer version.

Comment (exekias) [reply metricbeat/module/elasticsearch/_meta/env]: yes, it seems we use ES_HOST across all beats here: https://github.com/elastic/beats/blob/master/libbeat/tests/system/beat/beat.py#L546

**Pull request merged**